pkg/tfbridge/tests/provider_test.go

```diff
@@ -129,6 +130,363 @@ func TestInputsConfigModeEqual(t *testing.T) {
 	}
 }
 
+// TestStateFunc ensures that resources with a StateFunc set on their schema are correctly
+// handled. This includes ensuring that the PlannedPrivate blob is passed from
+// PlanResourceChange to ApplyResourceChange. If this is passed correctly, the provider
+// will see the original value of the field, rather than the value that was produced by
+// the StateFunc.
+func TestStateFunc(t *testing.T) {
+	t.Parallel()
+	crosstests.Create(t,
+		map[string]*schema.Schema{
+			"test": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				StateFunc: func(v interface{}) string {
+					return v.(string) + " world"
+				},
+			},
+		},
+		cty.ObjectVal(map[string]cty.Value{
+			"test": cty.StringVal("hello"),
+		}),
+		crosstests.InferPulumiValue(),
```

Comment:
Nit: would be cool if the interface was such that when omitted, it infers. So as not to spell this out. But perhaps spelling out is better as it calls attention to this.

Reply:
I prefer spelling it out, but you are now the second person to make that comment, so I'll reconsider. That will be in a separate PR.